src/engine/routing_algorithms/map_matching.cpp

```diff
@@ -201,7 +201,7 @@ SubMatchingList mapMatching(SearchEngineData<Algorithm> &engine_working_data,
                 prev_coordinate, current_coordinate);
             // assumes minumum of 4 m/s
             const EdgeWeight weight_upper_bound =
-                ((haversine_distance + max_distance_delta) / 4.) * facade.GetWeightMultiplier();
+                ((haversine_distance + max_network_distance) / 4.) * facade.GetWeightMultiplier();
```

Comment:
This would be only `max_network_distance` now, it basically assumes the weight encodes values in `second`.